### DISCUSSION TITLE
A Zod tool for Go+TypeScript users

### DISCUSSION THREAD

**Discussion by Southclaws** [colinhacks/zod#952] [Show and tell]
Hey all!

Love Zod for validation on the frontend! Most of my projects with a backend use Go for the API so I needed a way to validate responses from API calls to make sure from after a `fetch` call everything is type-safe.

Enter... [_Supervillain!_](https://github.com/Southclaws/supervillain/)

This small Go library takes a Go struct and generates a `z.object` using reflection. I use it as part of my build process to generate schemas and types for frontend code (since all my projects are monorepos so sharing this is easy)

I hope someone finds this useful! And if you have any suggestions or ideas, open an issue or PR ❤️

**Comment by JacobWeisenburger** [comment]
would you be willing to submit a PR to add this to the README in the Ecosystem section?

**Reply by Southclaws** [reply]
Yeah sure thing!
